checker/src/org/checkerframework/checker/initialization/InitializationAnnotatedTypeFactory.java

```diff
@@ -575,8 +608,18 @@ private void computeFieldAccessType(AnnotatedTypeMirror type,
                     .asType();
             boolean isInitializedForFrame = isInitializedForFrame(receiverType, fieldDeclarationType);
             if (isInitializedForFrame) {
-                type.replaceAnnotation(qualHierarchy.getTopAnnotation(UNCLASSIFIED));
+                // The receiver is initialized for this frame.
+                // Change the type of the field to @UnknownInitialization so that
+                // anything can be assigned to this field.
+                type.replaceAnnotation(UNCLASSIFIED);
+
+            } else if(computingAnnotatedTypeMirrorOfLHS) {
+                // The receiver is not initialized for this frame, and the type of a lhs is being computed.
+                // Only replace initialization annotation.
```

Comment:
It would be nice to explain the modifier change in this branch similarly to the previous branch.
If the explanation is the same, would it be easier to merge the branches?